osquery/logger/plugins/tls.cpp

```diff
@@ -144,20 +48,32 @@ static inline std::string genLogIndex(bool results, unsigned long& counter) {
          std::to_string(++counter);
 }
 
-Status TLSLoggerPlugin::logString(const std::string& s) {
-  if (stop_buffering) {
-    return Status(1, "Buffer is paused, dropping logs");
+static inline void amalgamate(std::vector<std::string>& input,
```

Comment:
If you feel as though having this other symbol which just loops with a sleep is worth the cognitive complexity, I would definitely prefer a different name. Nothing about this function requires anything to be "amalgamated", so it feels a bit misleading.